pindockers_test.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"path"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+)
+
+func TestDockerActions(t *testing.T) {
+	const inputDirectory = "./testfiles/pindockers/input"
+	const outputDirectory = "./testfiles/pindockers/output"
+	files, err := ioutil.ReadDir(inputDirectory)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	saveTr := Tr
+	defer func() { Tr = saveTr }()
+	Tr = httpmock.DefaultTransport
+
+	// add Table-Driven Tests
+	//Ping Docker Image
+	httpmock.RegisterResponder("GET", "https://ghcr.io/v2/",
+		httpmock.NewStringResponder(200, `{
+		}`))
+
+	httpmock.RegisterResponder("GET", "https://gcr.io/v2/",
+		httpmock.NewStringResponder(200, `{
+		}`))
+
+	//Fetch menifest file
+	//step.1> Get Token by this api call (https://auth.docker.io/token?service=<service>&scope=repository:<name>:<tag>)
+	//step.2> Get manifest file by setting header as (Authorization : Bearer <token>) and using this api call (https://<service>/v2/<name>/manifests/<tag>)
+	//step.3> Download response and save it in this path (testfiles/pindockers/response/<name>.txt)
+	httpmock.RegisterResponder("GET", "https://ghcr.io/v2/step-security/integration-test/int/manifests/latest",
```

Comment:
@Devils-Knight please find an example where docker hub is used to pull image. Docker hub is also popular, and if it can be used in such a context, we should support it. I think if the host name of registry is not specified, it might default to docker hub.